### PR TITLE
feat: use correct pkg manager when creating keystone app

### LINKED ISSUE #413: Don't use yarn internally
### Discussed in https://github.com/keystonejs/create-keystone-app/discussions/412

<div type='discussions-op-text'>

<sup>Originally posted by **RodrigoNovais** May  4, 2023</sup>
I've run `npm init keystone-app` to setup a new environment and it's dependencies where installed with yarn without my consent.  
Would be polite to ask the developers which  manager they want to use.  
I only have yarn installed in my machine cause I was obligated to use it in previours jobs and would like to avoid it as much as possible.  
Since the dependencies where automatically installed via yarn I don't even know if the CLI can initiate a project in a setup with pure npm. </div>

I don't think this is a matter of consent, as this is open source software - but we don't need to be using `yarn` specifically either, so it's something we could probably change.

@RodrigoNovais pull requests are accepted :yellow_heart:

### PR DESCRIPTION
**Pull request by unrevised6419**
Depending on what package manager tools were used, that package manager will be used.

Below package managers were tested, and are working:

- [x] NPM (npx, npm exec, npm init, npm create)
- [x] Yarn (yarn dlx, yarn create)
- [x] PNPm (pnpm exec, pnpm create, pnpm dlx)
- [x] Bun (bun create, bun init)

Closes #413

### CODE REVIEW

File: create-keystone-app/src/index.ts [hunk 1]
```diff
@@ -47,32 +47,24 @@ async function normalizeArgs(): Promise<Args> {
   };
 }
 
-const installDeps = async (cwd: string): Promise<'yarn' | 'npm'> => {
+function pkgManagerFromUserAgent(userAgent: string | undefined) {
+  if (!userAgent) return 'npm';
+  const pkgSpec = userAgent.split(' ')[0];
+  const [name, _version] = pkgSpec.split('/');
+  return name ?? 'npm';
+}
```
Comment (unrevised6419) [inline create-keystone-app/src/index.ts:55]: This is an adaptation of https://github.com/vitejs/vite/blob/2d50be2a5424e4f4c22774652ed313d2a232f8af/packages/create-vite/src/index.ts#L538-L546

**Comment by dcousens** [comment]
Hi @iamandrewluca,
Sorry I didn't get to reviewing this pull request.
I think I want to move this repository into https://github.com/keystonejs/keystone/ soon, as it's really difficult to keep the `create-keystone-app` up to date when it isn't synchronised with our monorepo.

**Comment by unrevised6419** [comment]
Hey @dcousens, I understand the pain of keeping it up to date. I'm here if you need any help from my side.

**Comment by unrevised6419** [comment]
Hey @dcousens, taking into consideration what you said about `create-keystone-app` above, are you also considering in the future releasing all mono-repo packages under the same version to keep them all in sync?

e.g.

@keystone-6/auth@7.1
@keystone-6/core@7.1

**Pull request closed without merging**